plotting/plot_window/base_qt_plot_window.py

```diff
@@ -0,0 +1,92 @@
+from PyQt4 import QtGui, QtCore
+from base_plot_window import BasePlotWindow
+from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
+from matplotlib.figure import Figure
+from matplotlib.patches import Rectangle
+
+class MatplotlibCanvas(FigureCanvas):
+    """Ultimately, this is a QWidget (as well as a FigureCanvasAgg, etc.)."""
+
+    def __init__(self, parent=None, width=5, height=4, dpi=100):
+        self.figure = Figure(figsize=(width, height), dpi=dpi)
+        self.axes = self.figure.add_subplot(111)
+        self.axes.hold(False)
+        FigureCanvas.__init__(self, self.figure)
+        self.setParent(parent)
+
+        FigureCanvas.setSizePolicy(self,
+                                   QtGui.QSizePolicy.Expanding,
+                                   QtGui.QSizePolicy.Expanding)
+        FigureCanvas.updateGeometry(self)
+        self._rect_cid = None
+        self._motion_cid = None
+        self._new_axes_limits = [[None,None],[None,None]] # For zooming [[x_lower,x_upper],[y_lower,y_upper]]
+        # TODO implement zoom history stack
+        self._zoom_history_stack = []
+        self._zoom_rect = None
+
+    def zoom_out(self):
```

Comment:
Is putting the zoom related functions in the canvas ok? I did that because most tools i needed were attributes of the canvas, although it would not be difficult to move this code to the window